src/core/common/message.cpp

```diff
@@ -649,11 +654,27 @@ uint16_t Message::ReadBytes(uint16_t aOffset, void *aBuf, uint16_t aLength) cons
     return static_cast<uint16_t>(bufPtr - reinterpret_cast<uint8_t *>(aBuf));
 }
 
+uint16_t Message::ReadBytes(const OffsetRange &aOffsetRange, void *aBuf) const
+{
+    return ReadBytes(aOffsetRange.GetOffset(), aBuf, aOffsetRange.GetLength());
+}
+
 Error Message::Read(uint16_t aOffset, void *aBuf, uint16_t aLength) const
 {
     return (ReadBytes(aOffset, aBuf, aLength) == aLength) ? kErrorNone : kErrorParse;
 }
 
+Error Message::Read(const OffsetRange &aOffsetRange, void *aBuf, uint16_t aLength) const
+{
+    Error error;
+
+    VerifyOrExit(aOffsetRange.Contains(aLength), error = kErrorParse);
+    error = (ReadBytes(aOffsetRange.GetOffset(), aBuf, aLength) == aLength) ? kErrorNone : kErrorParse;
```

Comment:
```suggestion
    Error error = kErrorNone;

    VerifyOrExit(aOffsetRange.Contains(aLength), error = kErrorParse);
    VerifyOrExit(ReadBytes(aOffsetRange.GetOffset(), aBuf, aLength) == aLength, error = kErrorParse);
```
What about writing like this?

Reply:
Good suggestion. Changed it in new push.